accepted/future-releases/static-extension-methods/feature-specification.md

```diff
@@ -46,214 +46,205 @@ extension MyFancyList<T> on List<T> {
 
-```dart
-extension TypedEquals<T> on T {
-  bool equals(T value) => this == value;
-}
-```
+You still cannot *export* two extensions with the same name. The rules for export makes it a compile-time error to add two declarations with the same name to the export scope of a library.
 
-Using such an extension as:
+##### Applicability
 
-```dart
-Object o = ...;
-String s = ...;
-print(s.equals(o));  // Compile-time type error.
-```
+An extension `E` is *applicable* to a simple or composite member invocation with corresponding member *basename* *m* and target expression `e`, where `e` has static type *S*, if
+
+- The invocation is an *instance* member invocation. That is the case if the expression `e` does not denote a prefix or a class, mixin or extension declaration *(then the member invocation would be a static invocation)*. An instance member invocation on `e` will always begin by evaluating `e` to an object, and then continue by performing an instance member invocation on that object.
+- The type *S*  does not have a member with the basename  *m*. For this, the types `dynamic`and `Never` are considered as having all member names, and an expression of type `void` cannot occur as the target of a member invocation, so none of these can ever have applicable extensions. Function types and the type `Function` are considered as having a `call` member. *This ensure that if there is an applicable extension, the existing invocation would otherwise be a compile-time error*. Members of `Object` exists on all types, so they can never be the target of implicit member invocations (they can also not be declared as extension members).
+- The extension application `E(x)` would be valid (not a compile-time error) where `x` is a fresh variable with static type *S* (to avoid type inference for any type parameters of `E` from affecting the already determined static type of `e`).
```

Comment:
Can't use `E(x)` if `E` may not be in scope (according to the current meaning of that phrase).

Reply:
added "in a scope where E denotes the extension". Then it does work.